### PR TITLE
Shared url session

### PR DESCRIPTION
**Pull request by antrix1989**

### CODE REVIEW

File: IdentityCore/src/network/MSIDHttpRequest.m [hunk 1]
```diff
@@ -44,8 +38,7 @@ - (instancetype)init
     
     if (self)
     {
-        _sessionConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
-        _session = [NSURLSession sessionWithConfiguration:_sessionConfiguration delegate:self delegateQueue:nil];
+        _sessionManager = MSIDUrlSessionManager.defaultManager;
```
Comment (oldalton) [inline IdentityCore/src/network/MSIDHttpRequest.m]: nit: can we use capital letters for URL?

File: IdentityCore/src/network/MSIDUrlSessionManager.m [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+- (instancetype _Nullable )initWithConfiguration:(nonnull NSURLSessionConfiguration *)configuration
+                                        delegate:(nullable MSIDUrlSessionDelegate *)delegate
+{
+    self = [super init];
+    if (self)
+    {
+        _configuration = configuration;
+        _session = [NSURLSession sessionWithConfiguration:configuration delegate:delegate delegateQueue:nil];
+        _delegate = delegate;
```
Comment (oldalton) [inline IdentityCore/src/network/MSIDUrlSessionManager.m]: why do we need to store delegate if we are already passing it to the session, which should retain it for the lifetime of the session?

File: IdentityCore/src/network/session_delegate/MSIDUrlSessionDelegate.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import <Foundation/Foundation.h>
+
+@interface MSIDUrlSessionDelegate : NSObject <NSURLSessionDelegate>
```
Comment (oldalton) [inline IdentityCore/src/network/session_delegate/MSIDUrlSessionDelegate.h]: this class seems to be empty...?

Comment (antrix1989) [reply IdentityCore/src/network/session_delegate/MSIDUrlSessionDelegate.h]: yes, it is empty right now, but we will add additional logic during client tls implementation.

**Pull request merged**